packages/sls-env/src/dependencies.ts

```diff
@@ -70,22 +95,33 @@ const resolveEventDependencies = <D, P, E, C>(
   context: C
 ): AppDependencyConverter<D> => {
   const updatedDependencies = Object.entries(dependencies).reduce((acc, [dependencyName, dependency]) => {
-    if (dependency?.type === 'EventDependency') {
-      const key = dependency.getKey(payload, event, context)
-      const eventDependency = dependency.dependencies[key]
-      if (!eventDependency) {
-        logger.warn(`could not extract event based dependency for ${dependencyName}`)
-        throw new EventDependencyError(`No event based dependency found for '${dependencyName}'`)
-      } else {
-        logger.debug(`replacing '${dependencyName}' with event based dependency`)
-        return { ...acc, [dependencyName]: eventDependency }
-      }
+    switch (dependency.type) {
+      case DependencyType.EventDependency:
+        const key = dependency.getKey(payload, event, context)
+        const eventDependency = dependency.dependencies[key]
+        if (!eventDependency) {
+          logger.warn(`could not extract event based dependency for ${dependencyName}`)
+          throw new EventDependencyError(`No event based dependency found for '${dependencyName}'`)
+        } else {
+          logger.debug(`replacing '${dependencyName}' with event based dependency`)
+          return { ...acc, [dependencyName]: eventDependency }
+        }
+      case DependencyType.DependencyFactory:
+        logger.debug(`replacing '${dependencyName}' with factory based dependency`)
+        return { ...acc, [dependencyName]: dependency.dependencyFactory({ payload, event, context }) }
```

Comment:
```suggestion
      case DependencyType.DependencyFactory:
        logger.debug(`replacing '${dependencyName}' with factory based dependency`)
        return { ...acc, [dependencyName]: dependency.create({ payload, event, context }) }
```

Reply:
```
type DependencyFactory<D, P = any, E = any, C = any> = {
  type: DependencyType.DependencyFactory
  dependencyFactory: DependencyFactoryResolver<D, P, E, C>
}
```
The `DependencyFactory` object has a method called `dependencyFactory` - did you mean for me to change that to `create` as changing it only here would not be enough.

I've named it `dependencyFactory` as that would be the actual factory provided by the user which will create the dependency. The `create` function makes sense within the context of this function, however if we look at the `DependencyFactory` object on its own we have
```
type: "DependencyFactory",
dependencyFactory: () => {}
```

Which in my opinion is a bit clearer of what it is that is is stored in that key
However you are right to suggest it should be a verb as it is actually a function which creates the dependency.

Perhaps `resolveDependency` ?
```
type DependencyFactory<D, P = any, E = any, C = any> = {
  type: DependencyType.DependencyFactory
  resolveDependency: DependencyFactoryResolver<D, P, E, C>
}
```